training/creating_dataset.py

```diff
@@ -129,3 +120,51 @@ def _create_df_with_basic_values(df: pd.DataFrame) -> pd.DataFrame:
     book = load_subcolumn_as_value("book")(get_subcolumn(df, "system/details/source"))
 
     return pd.concat([lvl, book], axis=1)
+
+
+def create_dataframe(
+    books: list[str] = DATASET_PATHS,
+    characteristics: list[str] = [
+        "system/abilities",
+        "system/attributes/ac",
+        "system/attributes/hp",
+    ],
+) -> pd.DataFrame:
+    """
+    Creates dataframe containing chosen characteristics, level (CR) and source book of monsters from chosen books
+
+    :param books: list of paths of books to load
+    :param characteristics: list of characteristics to load
+    :return: DataFrame with monsters (NPC) form chosen books and with chosen characteristics and their origin book
+    """
+    loading_methods = {
+        "system/abilities": move_values_level_up(value_name="mod"),
+        "system/attributes/ac": load_subcolumn_as_value("ac"),
+        "system/attributes/hp": load_subcolumn_as_value("hp"),
+        "system/attributes/perception": load_subcolumn_as_value("perception"),
+        "system/saves": move_values_level_up(value_name="value"),
+        "system/saves/fortitude": load_subcolumn_as_value("fortitude"),
+        "system/saves/reflex": load_subcolumn_as_value("reflex"),
+        "system/saves/will": load_subcolumn_as_value("will"),
+        "system/resources/focus": load_data_with_nan_val("focus", -1),
+    }
+    for i in range(len(books) - 1, -1, -1):
+        if not is_path_correct(books[i]):
+            books.pop(i)
+
+    bestiary = get_merged_bestiaries(books)
+    bestiary = bestiary[bestiary["type"] == "npc"]
+    df = _create_df_with_basic_values(bestiary)
+
+    for characteristic in characteristics:
+        if (loading_method := loading_methods.get(characteristic)) is not None:
+            part = loading_method(get_subcolumn(bestiary, characteristic))
+            df = pd.concat(
+                [
+                    df,
+                    part,
+                ],
+                axis=1,
+            )
```

Comment:
To jest niezła magia - rozumiem, co tu się dzieje, ale mam mocne wrażenie, że dałoby się to zrobić prościej. W skrócie - jaki jest tu problem, i jak go próbujecie rozwiązać? Bo wydaje mi się to lekko przekomplikowane

Reply:
Cechy "pozyskuje się" w nie koniecznie ten sam sposób z oryginalnego df, więc każda z dostępnych obecnie cech ma przyporządkowany sposób w jaki  się do niej dostać (dlatego też istnieją te funkcje zagnieżdżone) i potem sklejamy wszystkie kolumny razem. To było najlepsze na co wpadłam jak to pisałam miesiąc temu, ale przyjmę jakiś lepszy pomysł

Reply:
Podbijam temat, czy nie chcecie tego uprościć. Tzn. może tak zostać, to wasz kod i nic wam nie narzucam, ale moja rada jest taka, że im dalej, tym bardziej będzie wam to utrudniać życie, jakby dodawać kolejne cechy.

Reply:
@j-adamczyk 
Przepraszam, że zajęło to tyle czasu. 
Miałam nadzieję, iż wpadnę na to jak (niegłupio) zastosować ww. wektoryzację, ale nic nie wymyśliłam, więc jak na razie jest zwykły "for"

Reply:
Imo jest ciężko to zrobić inaczej niż teraz, bo te kolumny są mocno zagnieżdżone w sobie i każda ma inny sposób żeby się do niej dostać.